hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java

```diff
@@ -2180,6 +2218,9 @@ private void writeObject(ObjectOutputStream oos) throws IOException {
 
 		oos.defaultWriteObject();
 
+		oos.writeObject( exceptionMapper );
+		oos.writeObject( managedFlushChecker );
+		oos.writeObject( afterCompletionAction );
```

Comment:
we already write the SessionOwner to the stream later.  Writing these is completely unnecessary.